### PR TITLE
GDScript: Require type or variable initializer for export annotations

### LINKED ISSUE #74188: Using export_multiline should default to a blank string, not null
### Godot version

v4.0.stable.official [92bee43ad]

### System information

Windows 10

### Issue description

Using @export_multiline and then not defining the variable causes the inspector to show `<null>` when it should show nothing.

### Steps to reproduce

Make a script, add this line

`@export_multiline var dialog`

look in inspector window

### Minimal reproduction project

N/A

*Bugsquad edit:* Added mark up for `<null>` in the description, so GitHub does not show it as an empty string.

### PR DESCRIPTION
**Pull request by dalexeev**
Closes #74188.

### CODE REVIEW

File: modules/gdscript/tests/scripts/parser/features/export_enum.gd [hunk 1]
```diff
@@ -1,15 +1,13 @@
-@export_enum("Red", "Green", "Blue") var untyped
```
Comment (akien-mga) [inline modules/gdscript/tests/scripts/parser/features/export_enum.gd:1]: Might be good to add an `errors` test too with the expected failures for a few examples.

**Comment by dalexeev** [comment]
There are doubts about what behavior we want here. Perhaps annotations should modify the default value of a variable.

**Comment by vnen** [comment]
All the exports besides the default one already have a default type (the `export_annotation()` method has a template argument for it). I believe we could use it to set the default value, since all `@export_`  annotations have a specific type in mind. For instance: `@export_color_no_alpha`  and `@export_node_path` are very obvious, `file` and `dir` are expected to be strings, etc.

I believe this would be more intuitive behavior for users who do not have the habit of using static types. At first I thought this would be an extra burden to maintain, but since the types are already there we can use the information.

**Comment by dalexeev** [comment]
Okay, let's close this. But I haven't found a way to change the default value right now (I looked at `DataType`, `AssignableNode` and `VariableNode`).

**Pull request closed without merging**